app/routes/datasets/versions.py

```diff
@@ -240,13 +240,36 @@ async def append_to_version(
     # For the background task, we only need the new source uri from the request
     input_data = {"creation_options": deepcopy(default_asset.creation_options)}
     input_data["creation_options"]["source_uri"] = request.source_uri
+
+    # If source_driver is "text", this is a datapump request
+    if input_data["creation_options"]["source_driver"] != "text":
+        # Verify that source_driver is not None
+        if input_data["creation_options"]["source_driver"] is None:
+            raise HTTPException(
+                status_code=400,
+                detail="Source driver must be specified for non-datapump requests."
+            )
+
+        # Append the new layers to the existing ones
+        if input_data["creation_options"].get("layers") is None: # ERROR: layers is not defined
+            input_data["creation_options"]["layers"] = request.layers
+        elif request.layers is not None:
+            input_data["creation_options"]["layers"] += request.layers
+        else:
+            input_data["creation_options"]["layers"] = request.layers
+            
     background_tasks.add_task(
         append_default_asset, dataset, version, input_data, default_asset.asset_id
     )
 
     # We now want to append the new uris to the existing ones and update the asset
     update_data = {"creation_options": deepcopy(default_asset.creation_options)}
-    update_data["creation_options"]["source_uri"] += request.source_uri
+    update_data["creation_options"]["source_uri"] += request.source_uri # ERROR: only one source_uri is allowed
```

Comment:
I don't understand the ERROR comment.  Are you saying that you should add a check that len(request.source_uri) == 1 ?  Should probably put that check in or remove the comment?

Reply:
I've corrected this in later commits. Thanks for catching.